github/sendgrid/pr-thanks-by-sendgrid/rust/src/pr-thanks-by-sendgrid.rs

```diff
@@ -0,0 +1,53 @@
+use serde_json::json;
+use flows_connector_dsi::{github::inbound};
+use wasmedge_bindgen_macro::*;
+
+#[wasmedge_bindgen]
+pub fn run(s: String) -> Result<String, String> {
+    #[allow(unused_imports)]
+    use wasmedge_bindgen;
+    _run(s)
+}
+
+fn _run(s: String) -> Result<String, String> {
+    let payload = inbound(s)?;
+    
+    match payload.pull_request {
+        None => { return Err("no pull request found".to_string()); },
+        Some(_) => {},
+    }
+
+    let mut subject: &str = "";
+    let mut content = String::new();
+
+    if payload.action.unwrap() == "opened" {
```

Comment:
A better implementation is return early when the condition is not met. It should like:

```rust
if payload.get_action()? != "opened" {
    return Ok(String::new());
}
```

Reply:
thanks for your suggestions